pinot-core/src/test/java/org/apache/pinot/queries/MultiValueRawQueriesTest.java

```diff
@@ -338,6 +338,445 @@ public void testSelectQueries() {
+      String query = "SELECT mvRawIntCol, mvRawDoubleCol, mvRawStringCol from testTable GROUP BY mvRawIntCol, "
+          + "mvRawDoubleCol, mvRawStringCol ORDER BY mvRawStringCol LIMIT 10";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{
+          "mvRawIntCol", "mvRawDoubleCol", "mvRawStringCol"
+      }, new DataSchema.ColumnDataType[]{
+          DataSchema.ColumnDataType.INT, DataSchema.ColumnDataType.DOUBLE, DataSchema.ColumnDataType.STRING
+      });
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 10);
+
+      String stringVal = null;
+      for (int i = 0; i < 10; i++) {
+        Object[] values = recordRows.get(i);
+        assertEquals(values.length, 3);
+        String stringValueCur = (String) values[2];
+        if (stringVal != null && !stringValueCur.equals(stringVal)) {
+          assertTrue(stringValueCur.compareTo(stringVal) > 0);
+        }
+        stringVal = stringValueCur;
+      }
+    }
+    {
+      // Test a select with a VALUEIN transform function with group by
+      String query = "SELECT VALUEIN(mvRawIntCol, '0') from testTable WHERE mvRawIntCol IN (0) GROUP BY "
+          + "VALUEIN(mvRawIntCol, '0') LIMIT 10";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{"valuein(mvRawIntCol,'0')"},
+          new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.INT});
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 1);
+      Object[] values = recordRows.get(0);
+      assertEquals(values.length, 1);
+      int intRawVal = (int) values[0];
+      assertEquals(intRawVal, 0);
+    }
+    {
+      // Test a select with a VALUEIN transform function with group by order by
+      String query = "SELECT VALUEIN(mvRawDoubleCol, '0.0') from testTable WHERE mvRawDoubleCol IN (0.0) GROUP BY "
+          + "VALUEIN(mvRawDoubleCol, '0.0') ORDER BY VALUEIN(mvRawDoubleCol, '0.0') LIMIT 10";
+      ResultTable resultTable = getBrokerResponse(query).getResultTable();
+      assertNotNull(resultTable);
+      DataSchema dataSchema = new DataSchema(new String[]{"valuein(mvRawDoubleCol,'0.0')"},
+          new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.DOUBLE});
+      assertEquals(resultTable.getDataSchema(), dataSchema);
+      List<Object[]> recordRows = resultTable.getRows();
+      assertEquals(recordRows.size(), 1);
+      Object[] values = recordRows.get(0);
+      assertEquals(values.length, 1);
+      double doubleRawVal = (double) values[0];
+      assertEquals(doubleRawVal, 0.0);
+    }
+  }
+
+  @Test
+  public void testSelectWithFilterQueries() {
```

Comment:
As long as we have covered all the Mv column types and the filter operators =, IN, RANGE (<, > <=, >=), !=, this should be good.